pkg/models/resources/v1alpha3/interface.go

```diff
@@ -119,35 +122,46 @@ func DefaultObjectMetaFilter(item metav1.ObjectMeta, filter query.Filter) bool {
 		return strings.Contains(item.Name, string(filter.Value))
 		// /namespaces?page=1&limit=10&uid=a8a8d6cf-f6a5-4fea-9c1b-e57610115706
 	case query.FieldUID:
-		return strings.Compare(string(item.UID), string(filter.Value)) == 0
+		return string(item.UID) == string(filter.Value)
 		// /deployments?page=1&limit=10&namespace=kubesphere-system
 	case query.FieldNamespace:
-		return strings.Compare(item.Namespace, string(filter.Value)) == 0
+		return item.Namespace == string(filter.Value)
 		// /namespaces?page=1&limit=10&ownerReference=a8a8d6cf-f6a5-4fea-9c1b-e57610115706
 	case query.FieldOwnerReference:
 		for _, ownerReference := range item.OwnerReferences {
-			if strings.Compare(string(ownerReference.UID), string(filter.Value)) == 0 {
+			if string(ownerReference.UID) == string(filter.Value) {
 				return true
 			}
 		}
 		return false
 		// /namespaces?page=1&limit=10&ownerKind=Workspace
 	case query.FieldOwnerKind:
 		for _, ownerReference := range item.OwnerReferences {
-			if strings.Compare(ownerReference.Kind, string(filter.Value)) == 0 {
+			if ownerReference.Kind == string(filter.Value) {
 				return true
 			}
 		}
 		return false
 		// /namespaces?page=1&limit=10&annotation=openpitrix_runtime
 	case query.FieldAnnotation:
-		return labelMatch(item.Annotations, string(filter.Value))
-		// /namespaces?page=1&limit=10&label=kubesphere.io/workspace:system-workspace
+		return labelsMatch(item.Annotations, string(filter.Value))
+		// /namespaces?page=1&limit=10&label=kubesphere.io/workspace=system-workspace
 	case query.FieldLabel:
-		return labelMatch(item.Labels, string(filter.Value))
+		return labelsMatch(item.Labels, string(filter.Value))
 	default:
-		return false
+		// We should allow fields that are not found
+		return true
+	}
+}
+
+// labelsMatch handles multi label value pair split by ","
+func labelsMatch(labels map[string]string, filterStr string) bool {
+	filters := strings.SplitN(filterStr, ",", 2)
+	var match = true
+	for _, filter := range filters {
+		match = match && labelMatch(labels, strings.TrimSpace(filter))
```

Comment:
If the match is false, then it's not necessary to continue to try to match.

Reply:
Got it.